apps/devportal/data/markdown/pages/learn/accelerate/xm-cloud/implementation/redirect.md

```diff
@@ -0,0 +1,178 @@
+
+  return !!preValue || !!nextValue;
+};
+
+export default async function (req: NextRequest, ev: NextFetchEvent) {
+  if (!isPreview(req)) {
+    const reqUrl = new URL(req.url);
+    const hostName = reqUrl.hostname;
+
+    if (staticRedirects.hostName.indexOf(hostName) > -1) {
+      const urlPathName = decodeURIComponent(reqUrl.pathname).toLowerCase();
+
+      for (const redirectItem of staticRedirects.redirects) {
+        if (urlPathName === redirectItem.s) {
+          const destination = redirectItem.d;
+          if (destination.startsWith('http')) {
+            return NextResponse.redirect(new URL(destination), redirectItem.c);
+          } else {
+            return NextResponse.redirect(new URL(destination, req.url), redirectItem.c);
+          }
+        }
+      }
+    }
+  }
+}
+
+export const config = {
+  /*
+   * Match all paths except for:
+   * 1. /api routes
+   * 2. /_next (Next.js internals)
+   * 3. /sitecore/api (Sitecore API routes)
+   * 4. /- (Sitecore media)
+   * 6. all root files inside /public
+   */
+  matcher: ['/', '/((?!api/|_next/|sitecore/api/|-/).*)'],
+};
+```
+
+keep in mind the example above is a simplified version and you need to consider handling the multisite and caching in your implementation.
+
+## Discussion
+
+### When use the sxa redirects vs static redirects
+
+Before planning the redirect implementation, you need to ask few questions:
+
+- Should the redirects be editable by content authors or development team will handle it?
+- How many redirects are needed?
+- Can the number of redirects be reduced by using Regex?
+- When to use redirect map vs redirect item?
+
+If the redirects need to be editable by authors, you need to use the redirect or redirect map.
+
+If you have large number of redirects, which does not have a common pattern and can not be reduced using regex, you need to use your hosting provider's functions to manage it(ex. Vercel).
+
+When you need to map a simple path to another url for example `/about` to `/company/about` you can use the redirect item. The benefit if redirect item is the simplicity of the usage. A possible downside is the visibility and manageability in comparison with redirect map which groups and centeralized the redirercts. Because the redirect item is created as an item in the content tree, and when you want to see the list of existing redirects, you need to search for them.
+
+If you have a multiple redirects or you want to use regex you need to use redirect map instead. For example:
+
+```
+/company/products/arc-ow01 -> /products/oil/ow01
+/company/products/arc-ow02 -> /products/oil/ow02
+/company/products/arc-ow03 -> /products/oil/ow03
+```
+
+These type of redirects can be simplified with one regex statement in redirect map item:
+
+```
+/company/products/arc-(.*)$  -> /products/oil/$1
+```
+
+Again, always keep in mind the redirects are processed with every single request, if your regex is too complicated, it my cause a performance issue.
```

Comment:
Put this in an Alert panel and change slightly to:

> IMPORTANT: Always keep in mind the redirects are processed with every single request, if your regex is too complicated, it my cause a performance issue.

Reply:
Done